presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java

```diff
@@ -129,7 +129,16 @@ public QueuedStatementResource(
 
                             // forget about this query if the query manager is no longer tracking it
                             if (!dispatchManager.isQueryRegistered(entry.getKey())) {
-                                queries.remove(entry.getKey());
+                                Query query = queries.remove(entry.getKey());
+                                if (query != null) {
+                                    try {
+                                        query.destroy();
+                                    }
+                                    catch (Exception e) {
```

Comment:
This should either be `RuntimeException`, since `destroy()` doesn't throw checked exceptions, or be `Throwable` to guarantee we catch everything.